### PR TITLE
FIX invoice fetch not found syslog debug level instead of error

### PR DESCRIPTION
**Pull request by lvessiller-opendsi**
FIX invoice fetch not found syslog debug level instead of error

In invoice "fetch" method when a record was not found we had an error in syslog.
This not an error but a debug information.
And a free "SQL" resource was added .
And a syslog with "SQL" error was added.

### CODE REVIEW

File: htdocs/compta/facture/class/facture.class.php [hunk 1]
```diff
@@ -1730,14 +1730,17 @@ public function fetch($rowid, $ref = '', $ref_ext = '', $notused = '', $fetch_si
...
 			} else {
-				$this->error = 'Invoice with id='.$rowid.' or ref='.$ref.' or ref_ext='.$ref_ext.' not found';
-				dol_syslog(get_class($this)."::fetch Error ".$this->error, LOG_ERR);
+				dol_syslog(__METHOD__ . ' Invoice with id=' . $rowid . ' or ref=' . $ref . ' or ref_ext=' . $ref_ext . ' not found', LOG_DEBUG);
 				return 0;
 			}
 		} else {
-			$this->error = $this->db->error();
+			$this->error = $this->db->lasterror();
+			dol_syslog(__METHOD__ . ' Error : ' . $this->error, LOG_ERR);
```
Comment (eldy) [inline htdocs/compta/facture/class/facture.class.php:1743]: There is no need to add a dol_syslog here. When a sql error occurs, there is already automaticaly a dol_syslog to output the error.

File: htdocs/compta/facture/class/facture.class.php [hunk 2]
```diff
@@ -1730,14 +1730,17 @@ public function fetch($rowid, $ref = '', $ref_ext = '', $notused = '', $fetch_si
 					$this->error = $this->db->error();
 					return -3;
 				}
+
+				$this->db->free($resql);
+
 				return 1;
 			} else {
-				$this->error = 'Invoice with id='.$rowid.' or ref='.$ref.' or ref_ext='.$ref_ext.' not found';
```
Comment (eldy) [inline htdocs/compta/facture/class/facture.class.php:1735]: Why removing the set of ->error. It is required to return the reason of the fetch not found to the caller.

File: htdocs/compta/facture/class/facture.class.php [hunk 3]
```diff
@@ -1730,14 +1730,17 @@ public function fetch($rowid, $ref = '', $ref_ext = '', $notused = '', $fetch_si
...
-				dol_syslog(get_class($this)."::fetch Error ".$this->error, LOG_ERR);
```
Comment (eldy) [inline htdocs/compta/facture/class/facture.class.php:1736]: May be you can just replace with LOG_WARNING or LOG_DEBUG (In most cases, if we do a fetch, we are using an id and not founding the record is not the expected behaviour)

Comment (lvessiller-opendsi) [reply htdocs/compta/facture/class/facture.class.php:1736]: Ok I replaced with LOG_WARNING in syslog when no record was found

File: htdocs/compta/facture/class/facture.class.php (hunk 2 above)
Comment (lvessiller-opendsi) [reply htdocs/compta/facture/class/facture.class.php:1735]: @eldy
I don't understand why to keep an error in this class if it's not considered like an error (returning 0).
I think it's in the view (card) you show or not a message with a forbidden access.

File: htdocs/compta/facture/class/facture.class.php (hunk 1 above)
Comment (lvessiller-opendsi) [reply htdocs/compta/facture/class/facture.class.php]: Thanks for your remark, I removed this line.

File: htdocs/compta/facture/class/facture.class.php (hunk 2 above)
Comment (eldy) [reply htdocs/compta/facture/class/facture.class.php:1735]: Only > 0 is a success
If you have an id and want to fetch it, and you do not found it. This is not the expected behaviour. It is not a technical "error" (< 0) but error is not just technical error, it is not expected result.

**Pull request merged**